### DISCUSSION TITLE
Using BRouter with OsmAnd

### DISCUSSION THREAD

**Discussion by myfta** [osmandapp/OsmAnd#20649] [Maps Related]
I am trying to set up BRouter on an  Android phone. I have been using OsmAnd for a couple of years regularly. I wanted to use BRouter as in its native form (on the web app https://brouter.de/brouter-web/) you can set the profile to take gradients into account when routing.

I have BRouter installed and can see how to switch it on in OsmAnd. What I cannot find is how to select the profile settings in OsmAnd or tell OsmAnd which BRotre profile to use (i.e. the one that uses my gradient parameters).

Has anyone set this up?
Are there some clear instructions?

**Comment by Sonwon1** [comment]
BRouter needs better integration into OsmAnd. It is clunky having to manually select and download the routing zones. Can't we just integrate this with the maps downloaded?

**Comment by mikehgentry** [comment]
If you run the brouter app it allows you to set mappings for fast and short versions of car / foot / bicycle.

I think 'short' corresponds to not having 'fast' selected in OsmAnd, but it's a while since I've played around with it.

**Reply by myfta** [reply]
I could not see how fast/short are reflected in OsmAnd at all. The documentation is not at all clear either, unless I have not found it. 

It is not ideal, and not the way to do it, but I have edited the profile file directly on my device.

**Reply by mikehgentry** [reply]
There's a checkbox for 'fast' in Navigation Settings -> Route Parameters
